sdk/python/lib/test/runtime/test_localized_global_state.py

```diff
@@ -0,0 +1,16 @@
+import pytest
+from pulumi.runtime.settings import Settings
+
+@pytest.mark.asyncio
+async def test_settings():
+    default_settings_instance = Settings("project", "stack")
+
+    async def set_and_retrieve_settings_value(stack_name: str):
+        default_settings_instance.stack = stack_name
+        assert default_settings_instance.stack == stack_name
+        return default_settings_instance.stack
+
+    foo_name = await set_and_retrieve_settings_value("foo")
+    bar_name = await set_and_retrieve_settings_value("bar")
+    assert foo_name != bar_name != "project"
+    assert default_settings_instance.project == "project"
```

Comment:
If I have concerns about this PR, this test likely sums them up.  We will be isolating value changes to local contexts which will enable parallel pulumi programs to run without bleeding state.  However, any async tasks which depend on this formerly-global state will _also_ be isolated.  Tests pass with the current configuration but I think it's worth having a conversation here.

Reply:
I have verified that, at runtime for local stacks, settings values are preserved and work just fine, so it _could_ easily be weirdness with how our tests are implemented that cause us to need an escape hatch from the context-var-y-ness implemented in this PR.